### PR TITLE
Enable Falcon FP8 inference

### PR DESCRIPTION
**Pull request by schoi-habana**
falcon-7b, falcon-40b, falcon-180b quantization enabled with --reuse_cache

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

File: examples/text-generation/quantization_config/maxabs_measure_falcon.json
Comment (mandy-li) [inline examples/text-generation/quantization_config/maxabs_measure_falcon.json:1]: this should not be included in this PR, right? RnD will submit a PR for that.

File: optimum/habana/transformers/models/falcon/modeling_falcon.py [hunk 1]
```diff
@@ -52,6 +60,25 @@
...
+            training mode
+    """
+    out = F.dropout(x, p=prob, training=training)
+    out.add_(residual)
+    return out
+
+
 def apply_customized_rope(q, k, cos, sin, position_ids):
     if q.device.type == "hpu" and FusedRoPE:
         # TODO: remove `.clone()` when SynapseAI v1.15 is released
```
Comment (mandy-li) [inline optimum/habana/transformers/models/falcon/modeling_falcon.py:84]: remove this comment, or say something like: remove `.clone()` once the problem is fixed in SynapseAI

**Review by mandy-li — COMMENTED** [review]
Sun, please add the Falcon-180B fp8 inference command to README

@vivekgoe , please review the code and have your team to test FT with this PR.

**Review by vivekgoe — COMMENTED** [review]
@schoi-habana since the changes in modeling_falcon.py are large it is difficult to go over each of them. I would suggest you to request QA to check Falcon-40B & Falcon-180B fine-tuning with this PR to avoid any degrades.

Another generic comment, for the model classes you are overriding, please add in comment which transformer commit these are derived from and list the differences from that base commit. This will be useful when model is rebased to new transformer versions in future.

**Comment by mandy-li** [comment]
@libinta , since your team worked on Falcon-40B FT enablement, please review the code and test Falcon-40B FT as well. Thanks

File: examples/text-generation/README.md [hunk 2]
```diff
@@ -320,6 +320,37 @@ QUANT_CONFIG=./quantization_config/maxabs_quant_mixtral.json python run_generati
 --bf16 \
 --fp8
 ```
+
+Here is an example to measure the tensor quantization statistics on Falcon-180B with 8 cards:
+```bash
+QUANT_CONFIG=./quantization_config/maxabs_measure_include_outputs.json python ../gaudi_spawn.py \
+--use_deepspeed --world_size 8 run_lm_eval.py \
+-o acc_falcon180b_bs1_quant.txt \
+--model_name_or_path tiiuae/falcon-180B \
```
Comment (mandy-li) [inline examples/text-generation/README.md:329]: Falcon-180B is in gated repo. This command won't work

**Review by mandy-li — APPROVED** [review]

**Comment by astachowiczhabana** [comment]
https://github.com/huggingface/optimum-habana/pull/831
